sunpy/coordinates/tests/test_transformations.py

```diff
@@ -5,10 +5,50 @@
 from astropy.coordinates import SkyCoord, get_body_barycentric
 from astropy.time import Time
 
-from sunpy.coordinates import Helioprojective, HeliographicStonyhurst, HeliographicCarrington, get_sun_L0
+import sunpy.coordinates.transformations as trans
+from sunpy.coordinates import (Helioprojective, HeliographicStonyhurst,
+                               HeliographicCarrington, get_sun_L0,
+                               Heliocentric)
 from sunpy.time import parse_time
 
 
+def test_new_hcc_to_hgs():
+    # Generate 10 random coordinates
+    for i in range(50):
+        x = (np.random.rand() - 0.5) * u.km
+        y = (np.random.rand() - 0.5) * u.km
+        z = (np.random.rand() - 0.5) * u.km
+        time = '2007-05-04T21:08:12'
+        hcc = Heliocentric(x=x, y=y, z=z, obstime=time)
+        hgs_frame = HeliographicStonyhurst(obstime=time)
+
+        # New matrix based transformation
+        out1 = hcc.transform_to(hgs_frame)
+        # Old function based transformation
+        out2 = trans.old_hcc_to_hgs(hcc, hgs_frame)
+        # Check that they give the same results
+        assert_quantity_allclose(np.mod(out1.lon, 360 * u.deg),
+                                 np.mod(out2.lon, 360 * u.deg))
```

Comment:
The `np.mod` appears now, because the matrix based transform gives negative longitudes in some instances.

Reply:
`Heliocentric` should always have a wrap angle of 180 deg, so there shouldn't be a difference in "negative longitudes".  Put another way, if there *is* a difference between the two transforms in returned wrap angles, that's a bug that needs to be fixed.

Even if this were expected, the better approach (than using mod) would be to explicitly set both `Longitude` instances to the same `wrap_angle` and then check for equality.